# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/FetchUtil.cpp
# Commit: 682576598147
# Full Hash: 68257659814773dd5ce01d8dc9962eb3cd2313c9
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2022-08-03 03:42:48
# Regressor Bug: 1770630
# File Overlap Count: 1
# Description:
#   Bug 1770630 - Worker stream readers should contribute to busy count. r=dom-worker-reviewers,jstutte
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D149185
# ==============================================================================

diff -r 08d3ed24aab9 -r 682576598147 dom/fetch/FetchUtil.cpp
--- a/dom/fetch/FetchUtil.cpp	Wed Aug 03 00:26:29 2022 +0000
+++ b/dom/fetch/FetchUtil.cpp	Wed Aug 03 00:36:01 2022 +0000
@@ -295,14 +295,13 @@
     RefPtr<WorkerStreamOwner> self =
         new WorkerStreamOwner(aStream, std::move(target));
 
-    self->mWorkerRef = WeakWorkerRef::Create(aWorker, [self]() {
+    self->mWorkerRef = StrongWorkerRef::Create(aWorker, "JSStreamConsumer", [self]() {
       if (self->mStream) {
         // If this Close() calls JSStreamConsumer::OnInputStreamReady and drops
         // the last reference to the JSStreamConsumer, 'this' will not be
         // destroyed since ~JSStreamConsumer() only enqueues a release proxy.
         self->mStream->Close();
         self->mStream = nullptr;
-        self->mWorkerRef = nullptr;
       }
     });
 
@@ -326,7 +325,7 @@
   // Read from any thread but only set/cleared on the worker thread. The
   // lifecycle of WorkerStreamOwner prevents concurrent read/clear.
   nsCOMPtr<nsIAsyncInputStream> mStream;
-  RefPtr<WeakWorkerRef> mWorkerRef;
+  RefPtr<StrongWorkerRef> mWorkerRef;
   nsCOMPtr<nsIEventTarget> mOwningEventTarget;
 };
 